### ISSUE TITLE
 Example writing tests with map fields

### ISSUE THREAD

**Issue by abhipanda** [maoberlehner/vuex-map-fields#38]
I am using map fields in my vuex store but running the tests gives me the following error:

TypeError: this.$store.getters[getterType] is not a function

Are there any examples with vue-jest tests?

**Comment by Harti** [comment]
@abhipanda I encountered this issue when I forgot to set `namespaced: true` in one of my modules. Make sure that if you're using namespaces, you set them correctly.

**Comment by abhipanda** [comment]
@maoberlehner Yes I mock parts fo the app - my actual store is in the parent app where I have added the map fields functions.
@Harti - I don't see the issue while running the app but only when running tests

As a temporary workaround I resolved the issue by mocking the getter type and mapFields function

```
jest.mock('vuex-map-fields', () => ({
  getterType: jest.fn(),
  mapFields: jest.fn(),
}));
```

**Comment by maoberlehner** [comment]
The error occurs if the vuex-map-fields getter function can't find the property it's mapped to. I guess your mock store does not provide the property which the getter function should be mapped to?

**Comment by maoberlehner** [comment]
Thank you for reporting this issue.

May I ask you (@abhipanda and @n3ps) to please provide a [reduced test case](https://github.com/maoberlehner/vuex-map-fields/blob/master/CONTRIBUTING.md#reporting-issues). Otherwise it's pretty hard for me to do anything about it. My best guess is, that the mocked instance of your store has no `parent.someProperty` key. But it's just a guess and I can't do nothing more about it without seeing any code.

**Comment by blairdow** [comment]
> @maoberlehner Yes I mock parts fo the app - my actual store is in the parent app where I have added the map fields functions.
> @Harti - I don't see the issue while running the app but only when running tests
> 
> As a temporary workaround I resolved the issue by mocking the getter type and mapFields function
> 
> ```
> jest.mock('vuex-map-fields', () => ({
>   getterType: jest.fn(),
>   mapFields: jest.fn(),
> }));
> ```

Hey @maoberlehner I'm having the same issue with the getters throwing an error... where exactly would you put this jest.mock block? Sorry, I'm a bit new to vue test utils and jest.

**Comment by maoberlehner** [comment]
Hey @blairdow usually you put it as close to the code where you're relying on the mocked functionality as possible. In some cases it might be good to mock it globally for every test as it is described here: https://jestjs.io/docs/en/setup-teardown.html
